repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/do-upgrade.sh

```diff
@@ -194,6 +194,18 @@ ibdmp() {
     done
 }
 
+bring_up_network() {
+    if [ -f /etc/leapp-initram-network-manager ]; then
+        . /lib/dracut/hooks/cmdline/99-nm-config.sh
+        . /lib/dracut/hooks/initqueue/settled/99-nm-run.sh
```

Comment:
Please note that in general, you'd want these regenerated&executed with updated dracut, which also runs them - otherwise I'd rather use the ones from the updated package:

```
/lib/dracut/modules.d/35network-manager/nm-config.sh
/lib/dracut/modules.d/35network-manager/nm-run.sh
``` 

They also expect `/lib/nm-lib.sh` to exist (should be updated from the same directory).

But you should be able to bring up the system with either.

Reply:
So you'd prefer that we substitute L199-L200 with 
```
/lib/dracut/modules.d/35network-manager/nm-config.sh
/lib/dracut/modules.d/35network-manager/nm-run.sh
```
, right?

Reply:
Yes, that's what I assume, I've not tried it exists in the initrd... but I suppose it does. Let me know if you encounter issues, I'll investigate.

Reply:
@pvalena hmmm I have tried that on rhel8 vagrant box and got an early fail in initram stage http://pastebin.test.redhat.com/1077800

Reply:
Right, let me check where the files live, when in the initrd.

Reply:
Well, the locations are indeed where the files are installed in the initrd.

IIUC:

 - This initrd is built specially for the upgrade (the dracut version/build is corresponding to the system you're running)
 - you don't have access to the rootfs (or don't have it upgraded yet)
 - dracut+initqueue was invoked already, and you've reached some `target`
 - the network is still not running (or not properly configured) and you need to start the interfaces (NetworkManager) manually
 - ` Initqueue settled` means: 
    ``` This hook (initqueue/settled) gets executed every time udev has settled.```
 
(I'm sorry If I've missed something, I still need to take a look at the reproducer.)

In that case it would make sense to run those paths you had originally.

Reply:
@pvalena not sure I understand it right, but, will it work on systems where the network manager is not used?

Reply:
Yes, the `ifup` should do that.... TBH I'm not all that familiar with legacy networking, so I hope it's sufficient :).